kafka-rest/src/main/java/io/confluent/kafkarest/ConsumerManager.java

```diff
@@ -83,7 +83,7 @@ public ConsumerManager(KafkaRestConfig config, MetadataObserver mdObserver) {
     this.mdObserver = mdObserver;
     this.iteratorTimeoutMs = config.getInt(KafkaRestConfig.CONSUMER_ITERATOR_TIMEOUT_MS_CONFIG);
     this.workers = new Vector<ConsumerWorker>();
-    for (int i = 0; i < config.getInt(KafkaRestConfig.CONSUMER_THREADS_CONFIG); i++) {
+    for (int i = 0; i < config.getInt(KafkaRestConfig.CONSUMER_MAX_THREADS_CONFIG); i++) {
```

Comment:
Yeah, that's a good point. We also have to figure out what the behavior should be if -1 is used for max threads. How risky do you think the change from before to use the thread pool executor instead of the dedicated workers is?  We can keep all the existing locking unchanged at least.

Reply:
Shouldn't be any riskier than the v2 implementation. Do you see any potential complications?